autometa/common/external/samtools.py

```diff
@@ -47,32 +47,30 @@ def sort(sam, bam, nproc=mp.cpu_count()):
     Raises
     ------
     TypeError
-        nproc must be an interger greater than zero
+        nproc must be an integer greater than zero
     ChildProcessError
         Samtools did not run successfully, returns the return code from subprocessing call
     """
 
-    if type(nproc) is not int or nproc == 0:
+    if type(nproc) is not int or nproc <= 0:
         raise TypeError(f'nproc must be an integer greater than zero! Given: {nproc}')
     samtools_out_dir = os.path.dirname(os.path.abspath(bam))
-    outfile = os.path.basename(bam)
-    tempdir = tempfile.mkdtemp(suffix = None, prefix = 'samtools', dir = samtools_out_dir)
-    samtools_outfile = os.path.join(samtools_out_dir, outfile)
-    temp_outfile = os.path.join(tempdir, outfile)
-    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_outfile}'
-    logger.debug(f'cmd: {cmd}')
     samtools_err = os.path.join(samtools_out_dir, 'samtools.err')
     samtools_out = os.path.join(samtools_out_dir, 'samtools.out')
```

Comment:
This is a minor nitpick, but you could just name these variables `err` and `out`, because the entire script is within the context of samtools. This may improve readability.

```python
err = os.path.join(samtools_out_dir, 'samtools.err')
out = os.path.join(samtools_out_dir, 'samtools.out')
with open(out, 'w') as stdout, open(err, 'w') as stderr:
    ...
```

Reply:
But in case of some error, if the file is labeled ``samtools.err`` or ``samtools.out`` won't it help the user in identification?

Reply:
The logger will tell the user where the error occurred and the samtools files that were written may provide insight as to what went wrong in samtools